astropy/utils/data.py

```diff
@@ -1125,46 +1167,61 @@ def is_url_in_cache(url_key):
     Returns
     -------
     in_cache : bool
-        `True` if a download from ``url_key`` is in the cache
+        `True` if a download from ``url_key`` is in the cache, `False` if not
+        or if the cache does not exist at all.
+
+    See Also
+    --------
+    cache_contents : obtain a dictionary listing everything in the cache
     """
-    # The code below is modified from astropy.utils.data.download_file()
-    try:
-        dldir, urlmapfn = _get_download_cache_locs()
-    except OSError as e:
-        msg = 'Remote data cache could not be accessed due to '
-        estr = '' if len(e.args) < 1 else (': ' + str(e))
-        warn(CacheMissingWarning(msg + e.__class__.__name__ + estr))
-        return False
+    with _cache() as (dldir, url2hash):
+        return url_key in url2hash
 
-    with shelve.open(urlmapfn) as url2hash:
-        if url_key in url2hash:
-            return True
-    return False
 
+def cache_total_size():
+    """Return the total size in bytes of all files in the cache."""
+    with _cache() as (dldir, url2hash):
+        return sum(os.path.getsize(os.path.join(dldir, h))
+                   for h in url2hash.values())
```

Comment:
```suggestion
                   for h in url2hash.values()) * u.byte
```

Reply:
Then we can do things like this:
```python
>>> totsize.to(u.Mbyte)                                                    
<Quantity 7.749241 Mbyte>
>>> totsize.to(u.bit)                                                      
<Quantity 61993928. bit>
```

Reply:
Is this customary when we deal with bytes? I note human_file_size doesn't accept Quantities, just an integer number of bytes. In fact astropy/convolution/convolve.py strips the units off in one instance and provides a unit-free value in another.

Reply:
Depends on the context, usually. When user runs `cache_total_size`, it is very unlikely they will route the output to something else. And for the rare case when they do, they can strip off unit like passing in `.value` attribute.

The reason I bought this up is that when I was going through your user documentation and try running the new functions, I found myself wondering out its output unit and had to look up the API doc. If it returns Quantity, then I would have known right away.

Reply:
> Ops, it seems like conflicts came from #8237 changes to `data.py` and its tests. I hope resolving the conflicts won't be too painful for you. Please let me know if you need help on this. Sorry for any inconvenience caused!

Should be okay. Uh, I'll try to sort out the other stuff first so the rebase won't trample all your commments.

Reply:
Mmm. But one might expect cache_total_size to be comparable to get_free_space_in_dir, which we immediately compare to the size of files in check_free_space_in_dir. So presumably one would change all those files or none. We also use get_free_space_in_dir in astropy/io/fits/util.py.

Reply:
Ah, okay, good point. Let's leave it unitless then. It was just me being lazy to manually convert to MB, heh heh.